modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1114,52 +1114,14 @@ public TableImpl table(IgniteUuid id) throws NodeStoppingException {
      * @param checkConfiguration True when the method checks a configuration before tries to get a table, false otherwise.
      * @return A table or {@code null} if table does not exist.
      */
-    private CompletableFuture<Table> tableAsync(String name, boolean checkConfiguration) {
+    private CompletableFuture<? extends Table> tableAsync(String name, boolean checkConfiguration) {
         if (checkConfiguration && !isTableConfigured(name)) {
             return CompletableFuture.completedFuture(null);
         }
 
-        Table tbl = tables.get(name);
+        ExtendedTableView view = (ExtendedTableView) ConfigurationUtil.directValue(tablesCfg.tables()).get(name);
```

Comment:
Please move this to another method, which return ID or null if table dos not found.